src/core/lbgpu_cuda.cu

```diff
@@ -4265,4 +4268,65 @@ void lb_lbfluid_get_population( int xyz[3], float population_host[LBQ], int c )
 }
 
 
+/**
+ * get_fluid_velocity_at_particle_positions_kernel
+ * (Eq. (12) Ahlrichs and Duenweg, JCP 111(17):8225 (1999))
+ * @param n_a                   Pointer to local node residing in array a (Input)
+ * @param *particle_data        Pointer to the particle position and velocity (Input)
+ * @param *u_gpu                Pointer to float array (Output)
+ * @param node_f                Struct for node force (Input)
+ * @param *d_v                  Pointer to local device values
+ * @param coupling              int denoting wether to use two- (0) or three-point (1) coupling for the velocity interpolation (Input)
+*/
+__global__ void lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel(LB_nodes_gpu n_a, CUDA_particle_data *particle_data, float* u_gpu, LB_node_force_gpu node_f, LB_rho_v_gpu *d_v, int coupling){
+  unsigned int index = blockIdx.y * gridDim.x * blockDim.x + blockDim.x * blockIdx.x + threadIdx.x;
+  if (index < para.number_of_particles) {
+    float position[3];
+    unsigned int node_index[8];
+    float mode[19*LB_COMPONENTS];
+    float delta[8];
+    position[0] = particle_data[index].p[0];
+    position[1] = particle_data[index].p[1];
+    position[2] = particle_data[index].p[2];
+    // Do the velocity interpolation
+    if (coupling == 0) {
+        interpolation_two_point_coupling(n_a, position, node_index, mode, nullptr, delta, &u_gpu[3*index]);
+    } else if (coupling == 1) {
+        interpolation_three_point_coupling(n_a, position, node_index, nullptr, delta, &u_gpu[3*index]);
+    }
+  }
+}
+
+/** interface to get fluid velocities at particle positions
+*/
+std::vector<float> lb_lbfluid_get_fluid_velocity_at_particle_positions(std::string coupling="twopoint") {
```

Comment:
Please move the string parsing to a sep function or to python and raise an error if it is not one of the allowed values. The current impl will just return garbage if you do lb_lbfluid_get_fluid_velocity_at_particle_positions("not_a_coupling").